source/Camera/InfoUtilOther/C++Halcon/HalconConnectUsingGenICam/HalconConnectUsingGenICam.cpp

```diff
@@ -0,0 +1,102 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string filename)
+{
+	const HalconCpp::HString filetype{ "ply" };
+	const HalconCpp::HString filenameCstring{ filename.c_str() };
+	const HalconCpp::HString genParamName{ "invert_normals" };
+	const HalconCpp::HString genParamValue{ "false" };
+	model.WriteObjectModel3d(filetype, filenameCstring, genParamName, genParamValue);
+}
+
+
+HalconCpp::HObjectModel3D setColorInObjectModel3D(HalconCpp::HImage RGB, HalconCpp::HObjectModel3D pcHalcon) {
+
+	HalconCpp::HImage R = RGB.AccessChannel(1);
+	HalconCpp::HImage G = RGB.AccessChannel(2);
+	HalconCpp::HImage B = RGB.AccessChannel(3);
+
+	HalconCpp::HRegion domain = R.GetDomain();
+	HalconCpp::HTuple rows, cols;
+	domain.GetRegionPoints(&rows, &cols);
+
+	HalconCpp::HTuple Rtup = R.GetGrayval(rows, cols);
+	HalconCpp::HTuple Gtup = G.GetGrayval(rows, cols);
+	HalconCpp::HTuple Btup = B.GetGrayval(rows, cols);
+
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+
+	return pcHalcon;
+}
+
+
+int main()
+{
+	try
+	{
+		std::cout << "Connecting to camera" << std::endl;
+
+		// Connect to camera using GeniCamTL driver
+		HalconCpp::HFramegrabber Framegrabber;
+		Framegrabber.OpenFramegrabber("GenICamTL", 1, 1, 0, 0, 0, 0, "progressive", -1, "default", -1, "false", "default", "default", 0, 0);
+
+		// Configure 3D-settings
+		Framegrabber.SetFramegrabberParam("create_objectmodel3d", "enable");
+		Framegrabber.SetFramegrabberParam("add_objectmodel3d_overlay_attrib", "enable");
+		Framegrabber.SetFramegrabberParam("AcquisitionMode", "SingleFrame");
+
+		// Set camera settings
+		Framegrabber.SetFramegrabberParam("Iris", 20);
+		Framegrabber.SetFramegrabberParam("ExposureTime", 8333);
+		Framegrabber.SetFramegrabberParam("Gain", 2);
+		Framegrabber.SetFramegrabberParam("Brightness", 1.0);
+
+		Framegrabber.SetFramegrabberParam("OutlierFilterEnabled", 1);
+		Framegrabber.SetFramegrabberParam("GaussianFilterEnabled", 1);
+		Framegrabber.SetFramegrabberParam("GaussianFilterSigma", 1.5);
+		Framegrabber.SetFramegrabberParam("OutlierFilterThreshold", 5);
+
+		// Capture frame
+		std::cout << "Capturing frame" << std::endl;
+
+		HalconCpp::HRegion region;
+		HalconCpp::HXLDCont contours;
+		HalconCpp::HTuple data;
+		HalconCpp::HImage frame = Framegrabber.GrabData(&region, &contours, &data);
+
+		// Construct HObjectModel3D from x,y,z data
+		HalconCpp::HImage X = frame.SelectObj(1);
```

Comment:
These could be `const auto X = frame.SelectObj(1);`, again I don't know what is the preferred method. @eskaur ?

Reply:
If X, Y, Z, RGB are not going to be changed `const` can be used regardless.